lib/SILOptimizer/FunctionSignatureTransforms/ExistentialTransform.cpp

```diff
@@ -0,0 +1,547 @@
+    ThunkBody->createFunctionArgument(ArgDesc.Arg->getType(), ArgDesc.Decl);
+  }
+
+  /// Builder to add new instructions in the Thunk.
+  SILBuilder Builder(ThunkBody);
+  SILOpenedArchetypesTracker OpenedArchetypesTracker(F);
+  Builder.setOpenedArchetypesTracker(&OpenedArchetypesTracker);
+  Builder.setCurrentDebugScope(ThunkBody->getParent()->getDebugScope());
+
+  /// Location to insert new instructions.
+  SILLocation Loc = ThunkBody->getParent()->getLocation();
+
+  /// Create the function_ref instruction to the NewF.
+  FunctionRefInst *FRI = Builder.createFunctionRef(Loc, NewF);
+
+  auto GenCalleeType = NewF->getLoweredFunctionType();
+  auto CalleeGenericSig = GenCalleeType->getGenericSignature();
+  auto OrigGenCalleeType = F->getLoweredFunctionType();
+  auto OrigCalleeGenericSig = OrigGenCalleeType->getGenericSignature();
+
+  /// Determine arguments to Apply.
+  /// Generate opened existentials for generics.
+  llvm::SmallVector<SILValue, 8> ApplyArgs;
+
+  /// Ideally this should be a SubstitutionMap. TODO.
+  llvm::SmallDenseMap<GenericTypeParamType *, Type> Generic2OpenedTypeMap;
+  for (auto &ArgDesc : ArgumentDescList) {
+    auto iter = Arg2GenericTypeMap.find(ArgDesc.Index);
+    if (iter != Arg2GenericTypeMap.end()) {
+      ArchetypeType *Opened;
+      SILValue OrigOperand = ThunkBody->getArgument(ArgDesc.Index);
+      auto SwiftType = ArgDesc.Arg->getType().getASTType();
+      auto OpenedType =
+          SwiftType->openAnyExistentialType(Opened)->getCanonicalType();
+      auto OpenedSILType = NewF->getModule().Types.getLoweredType(OpenedType);
+      SILValue archetypeValue;
+      auto ExistentialRepr =
+          ArgDesc.Arg->getType().getPreferredExistentialRepresentation(M);
+      switch (ExistentialRepr) {
+      case ExistentialRepresentation::Opaque: {
+        archetypeValue = Builder.createOpenExistentialAddr(
+            Loc, OrigOperand, OpenedSILType,
+            (ArgDesc.Arg->hasConvention(
+                 SILArgumentConvention::Indirect_Inout) ||
+             ArgDesc.Arg->hasConvention(
+                 SILArgumentConvention::Indirect_InoutAliasable))
+                ? OpenedExistentialAccess::Mutable
+                : OpenedExistentialAccess::Immutable);
+        ApplyArgs.push_back(archetypeValue);
+        break;
+      }
+      case ExistentialRepresentation::Class: {
+        archetypeValue =
+            Builder.createOpenExistentialRef(Loc, OrigOperand, OpenedSILType);
+        ApplyArgs.push_back(archetypeValue);
+        break;
+      }
+      default: {
+        assert(true);
+        break;
+      }
+      };
+      Generic2OpenedTypeMap.insert(
+          std::pair<GenericTypeParamType *, Type>(iter->second, OpenedType));
+    } else {
+      ApplyArgs.push_back(ThunkBody->getArgument(ArgDesc.Index));
+    }
+  }
+
+  unsigned int OrigDepth = 0;
+  if (F->getLoweredFunctionType()->isPolymorphic()) {
+    OrigDepth = OrigCalleeGenericSig->getGenericParams().back()->getDepth();
```

Comment:
Make this OrigDepth = OrigCalleeGenericSig->getGenericParams().back()->getDepth()+1;
and then remove the check for polymorphic on line 457 to F->getLoweredFunctionType()->isPolymorphic() and then the 458 is now < not <=